docs/hugo/content/contributing/generator-overview.md

```diff
@@ -29,44 +29,46 @@ The size of each dot reflects the size of the file; the legend in the corner sho
 
-When a `Type` is given a `TypeName`, it becomes a `TypeDefinition` and can be emitted as the source code for a Go type definition. A set of many `TypeDefinition`, each with a unique name is a `TypeDefinitionSet`. 
+This list is not exhaustive; other implementations of `Type` are used within limited scopes. For example, [`AllOfType`](https://github.com/Azure/azure-service-operator/blob/main/v2/tools/generator/internal/astmodel/allof_type.go) and [`OneOfType`](https://github.com/Azure/azure-service-operator/blob/main/v2/tools/generator/internal/astmodel/oneof_type.go) represent JSON Schema techniques for creating object definitions via composition.
 
-Both `ResourceType` and `ObjectType` act as containers, each implementing `PropertyContainer`, `FunctionContainer`, and `TestCaseContainer`. These do pretty much what you'd expect from the names, though the implementations differ between `ResourceType` and `ObjectType`. For example, where an `ObjectType` implements `PropertyContainer` by providing support for an arbitrary set of properties, `ResourceType` has only `Spec` and `Status`.
+Usefully, there is also [`TypeName`](https://github.com/Azure/azure-service-operator/blob/main/v2/tools/generator/internal/astmodel/type_name.go) which is both a type in itself and an indirect reference to a type defined elsewhere.
 
-Most implementations of `astmodel.Function` are found in the **functions** package. New function implementations should go here; existing implementations are slowly being relocated.
+When a `Type` is given a `TypeName`, it becomes a [`TypeDefinition`](https://github.com/Azure/azure-service-operator/blob/main/v2/tools/generator/internal/astmodel/type_definition.go) and can be emitted as the source code for a Go type definition. A set of many `TypeDefinition`, each with a unique name is a [`TypeDefinitionSet`](https://github.com/Azure/azure-service-operator/blob/main/v2/tools/generator/internal/astmodel/types.go). 
 
-All implementations of `astmodel.TestCase` are found in the **testcases** package. New test case implementations should go here.
+Both `ResourceType` and `ObjectType` act as containers, each implementing [`PropertyContainer`](https://github.com/Azure/azure-service-operator/blob/main/v2/tools/generator/internal/astmodel/property_container.go), and [`FunctionContainer`](https://github.com/Azure/azure-service-operator/blob/main/v2/tools/generator/internal/astmodel/function_container.go). These do pretty much what you'd expect from the names, though the implementations differ between `ResourceType` and `ObjectType`. For example, where an `ObjectType` implements `PropertyContainer` by providing support for an arbitrary set of properties, `ResourceType` has only `Spec` and `Status`.
```

Comment:
minor: Could also mention `InterfaceImplementer` here. It's not an interface (more a mixin sorta thing), but it's common to both and allows them to specify that they implement a particular interface (including the methods for that interface and a type-assertion about it).

Reply:
Good idea. Done.